test/framework/dump.go

```diff
@@ -209,6 +210,37 @@ func (f *GardenerFramework) dumpGardenerExtension(extension v1alpha1.Object) {
 	}
 }
 
+func (f *CommonFramework) DumpLogsForPodsWithLabelsInNamespace(ctx context.Context, ctxIdentifier string, k8sClient kubernetes.Interface, namespace string, labels client.MatchingLabels) error {
```

Comment:
Nit: you could make it 1 idea more generic:

```suggestion
func (f *CommonFramework) DumpLogsForPodsWithLabelsInNamespace(ctx context.Context, ctxIdentifier string, k8sClient kubernetes.Interface, namespace string, opts ...client.ListOption) error {
```

Reply:
Good idea, thanks!